### ISSUE TITLE
[Bug]: Patch Circle not plotted correct in the symlog scale

### ISSUE THREAD

**Issue by lrtfm** [matplotlib/matplotlib#22901]
### Bug summary

Patch Circle is not plotted correctly in the `symlog` scale.

### Code for reproduction

```python
from matplotlib.path import Path
import matplotlib.pyplot as plt
import numpy as np

circle1 = plt.Circle((0, 1), 1, fill=False)

path = circle1.get_path()

g = list(path.iter_bezier())[1][0]

ts = np.linspace(0, 1, 100)
d = []
xx = []
yy = []
for t in ts:
    v = g.point_at_t(t)
    xx.append(v[0])
    yy.append(v[1] + 1)
    d.append(np.sqrt(1 - v[0]**2) - np.abs(v[1]))

fig, ax = plt.subplots()
ax.add_patch(circle1)
ax.plot(xx, yy, 'b')
ax.set_xscale('symlog', linthresh=1e-1)
ax.set_yscale('symlog', linthresh=1e-5)

plt.savefig('patch.pdf', bbox_inches='tight')
```

### Actual outcome

![1650975844(1)](https://user-images.githubusercontent.com/1192812/165299201-6799107d-6688-4ca1-bfca-d5c5537aaf6a.png)

### Expected outcome

The blue line should coincide with the black line.

### Matplotlib Version

3.5.1

**Comment by lrtfm** [comment]
Yes, the bezier line is accurate enough. In  your figure (the plot of d), we can see the max error of the bezier line is below 1e-5. However, in my figure, the error is bigger then (1e-3 - 1e-5). This is weird. 
When I create a circle by using the following formula (cos and sin), we will see the same result as the bezier line.
```
from matplotlib.path import Path
import matplotlib.pyplot as plt
import numpy as np

circle1 = plt.Circle((0, 1), 1, fill=False)

theta = np.linspace(0, 2*np.pi, 1000)
xx =  np.cos(theta)
yy = 1 + np.sin(theta)

fig, ax = plt.subplots()
ax.add_patch(circle1)
ax.plot(xx, yy, 'b')
ax.set_xscale('symlog', linthresh=1e-1)
ax.set_yscale('symlog', linthresh=1e-5)
```
![183679ED-7E18-4588-8398-43031C94DC8D](https://user-images.githubusercontent.com/1192812/165800161-83c93783-b00c-49d0-88ff-19eaa06a6b60.jpeg)

**Comment by tacaswell** [comment]
oooh, 🤦🏻 I get it now 🐑 

I think the overall technical explanation is still correct (in the case of the `Circle` we are transforming the control points and then doing the interpolation in screen space vs sampling first and then transforming).  Doing the transfrom than sample approach is not correctly accounting from the transition from log to linear (because it only looks at the points where it has values and is blind to anything in the middle) where are directly sampling the Bezier curve or using the sin/cos you have enough samples in data space that you correctly see the transitions.

I do not spend a lot of time looking at symlog plots and am still very confused by the inversion of curvature is right....

**Comment by lrtfm** [comment]
@tacaswell thanks for your explanation. I agree with that. Could you explain the meaning of "the inversion of curvature is right"?

**Comment by timhoffm** [comment]
Let‘s close this. Circle documents that it‘s a bezier-based approximation. That’s good enough and I don‘t think we can reasonably do anything to improve this.
